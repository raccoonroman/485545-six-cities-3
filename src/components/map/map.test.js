import React from "react";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Map from "./map.jsx";


configure({adapter: new Adapter()});

const offers = [
  {
    id: 10,
    offerInfo: {
      title: `Title 10`,
      price: 105,
      category: `house`,
      rating: 4,
      imageUrl: `https://i.picsum.photos/id/10/260/200.jpg`,
      coords: [52.3909553943508, 4.929309666406198],
    }
  },
  {
    id: 11,
    offerInfo: {
      title: `Title 11`,
      price: 125,
      mark: `some mark`,
      category: `hotel`,
      rating: 3.6,
      imageUrl: `https://i.picsum.photos/id/11/260/200.jpg`,
      coords: [52.3909553943508, 4.85309666406198],
    }
  },
  {
    id: 12,
    offerInfo: {
      title: `Title 12`,
      price: 160,
      category: `apartment`,
      rating: 4.6,
      imageUrl: `https://i.picsum.photos/id/12/260/200.jpg`,
      coords: [52.369553943508, 4.85309666406198],
    }
  },
];


it(`Should <Map /> render correctly`, () => {
  const div = global.document.createElement(`div`);
  global.document.body.appendChild(div);
  const tree = mount(<Map offers={offers} />, {attachTo: div});

  expect(tree.getDOMNode()).toMatchSnapshot();
});
