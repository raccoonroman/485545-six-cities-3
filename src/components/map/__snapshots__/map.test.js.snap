// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should <Map /> render correctly 1`] = `
<section
  class="cities__map map"
>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://c.basemaps.cartocdn.com/rastertiles/voyager/12/2371/1402.png"
              style="width: 256px; height: 256px; left: -229px; top: -153px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -69303px; top: -14391px; z-index: -14391;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin-active.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -68638px; top: -14327px; z-index: -14327;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -68792px; top: -14411px; z-index: -14411;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
